experimental/dds/tree2/src/feature-libraries/editable-tree-2/lazyTree.ts

```diff
@@ -329,17 +330,17 @@ export class LazyMap<TSchema extends MapSchema>
 	}
 
 	public forEach(
-		callbackFn: (
-			value: UnboxField<TSchema["mapFields"]>,
-			key: FieldKey,
-			map: MapNode<TSchema>,
-		) => void,
-		thisArg?: any,
+		callbackFn: (value: UnboxField<TSchema["mapFields"]>, key: FieldKey, map: this) => void,
 	): void {
-		const fn = thisArg !== undefined ? callbackFn.bind(thisArg) : callbackFn;
-		for (const [key, value] of this.entries()) {
-			fn(value, key, this);
-		}
+		forEachField(this[cursorSymbol], (cursor) =>
+			callbackFn(
+				unboxedField(this.context, this.schema.mapFields, cursor) as UnboxField<
+					TSchema["mapFields"]
+				>,
+				cursor.getFieldKey(),
+				this,
+			),
+		);
 	}
```

Comment:
@noencke - If it's helpful as a starting point, I have a never-committed suite of tests from when I added Array-compatible read/write APIs to our previous editable sequence field [here](https://github.com/microsoft/FluidFramework/blob/435af1a07f1903e49ac024721ce896d9a1b75729/experimental/dds/tree2/src/test/feature-libraries/editable-tree/editableTree.arrayLike.spec.ts).

Or if you like, I'm happy for you to commit and I can follow up with compatibility tests.

Reply:
That's awesome! We should definitely incorporate the tests for the methods that we keep (e.g. indexOf, map, reduce). And maybe it is valuable to check that all the Array methods work on lists/sequences via `Array.prototype...call(list)` as well.